scripts/dotnet-cli-build/PublishTargets.cs

```diff
@@ -110,6 +120,38 @@ public static BuildTargetResult PublishDebFileToDebianRepo(BuildTargetContext c)
         }
 
         [Target]
+        [BuildPlatforms(BuildPlatform.Ubuntu)]
+        public static BuildTargetResult PublishSharedFrameworkDebToDebianRepo(BuildTargetContext c)
+        {
+            var packageName = Monikers.GetDebianSharedFrameworkPackageName(c);
+            var installerFile = c.BuildContext.Get<string>("SharedFrameworkInstallerFile");
+            var uploadUrl = $"https://dotnetcli.blob.core.windows.net/dotnet/{Channel}/Installers/{Version}/{Path.GetFileName(installerFile)}";
+            var uploadJson = GenerateUploadJsonFile(packageName, Version, uploadUrl);
+
+            Cmd(Path.Combine(Dirs.RepoRoot, "scripts", "publish", "repoapi_client.sh"), "-addpkg", uploadJson)
+                    .Execute()
+                    .EnsureSuccessful();
+
+            return c.Success();
+        }
+
+        [Target]
+        [BuildPlatforms(BuildPlatform.Ubuntu)]
+        public static BuildTargetResult PublishSharedHostDebToDebianRepo(BuildTargetContext c)
+        {
+            var packageName = Monikers.GetDebianSharedHostPackageName(c);
+            var installerFile = c.BuildContext.Get<string>("SharedHostInstallerFile");
+            var uploadUrl = $"https://dotnetcli.blob.core.windows.net/dotnet/{Channel}/Installers/{Version}/{Path.GetFileName(installerFile)}";
+            var uploadJson = GenerateUploadJsonFile(packageName, Version, uploadUrl);
+
+            Cmd(Path.Combine(Dirs.RepoRoot, "scripts", "publish", "repoapi_client.sh"), "-addpkg", uploadJson)
+                    .Execute()
+                    .EnsureSuccessful();
+
+            return c.Success();
+        }
+
```

Comment:
I agree this whole file could stand for some factoring in a separate PR. :smile: 

Intention here is to quickly bring up the scenario, we have a number of places where code improvements are needed but I'd prefer to have something working to iterate on.